llvm/docs/CallGraphSection.md

```diff
@@ -0,0 +1,29 @@
+# .callgraph Section Layout
+
+The `.callgraph` section is used to store call graph information for each function, which can be used for post-link analyses and optimizations. The section contains a series of records, with each record corresponding to a single function.
+
+For efficiency, we make a distinction between direct and indirect call data. For direct calls, we record the unique callees, not the location of each individual call. For indirect calls, we record the location of each call site and the type ID of the callee. Post link analysis scripts which utilize this information to reconstuct the program call graph can potentially receive more information regarding  indirect callsites from the user to improve the precision of the call graph.
```

Comment:
```suggestion
For efficiency, we make a distinction between direct and indirect call data. For direct calls, we record the unique callees, not the location of each individual call. For indirect calls, we record the location of each call site and the type ID of the callee. Users of this information can potentially derive more information regarding  indirect callsites (e.g. from the user) to improve the precision of the call graph.
```
Maybe something like this to avoid post link verbiage? its really just any consumer of this data, right?

Reply:
This difference between indirect and direct call handling does not exist anymore. Dropped this paragraph.